pkg/v1/transport/books/transport.go

```diff
@@ -0,0 +1,92 @@
+package books
+
+import (
+    "bytes"
+    "context"
+    "encoding/json"
+    "errors"
+    "io"
+    "net/http"
+
+    svc "github.com/King-Rullmann-Book-Club/books-api/pkg/v1/service/books"
+    ep "github.com/King-Rullmann-Book-Club/books-api/pkg/v1/endpoints/books"
+
+    "github.com/gorilla/mux"
+    httptransport "github.com/go-kit/kit/transport/http"
+)
+
+// NewTransport returns a new mux router for exposed endpoints
+func NewTransport(s svc.Service) http.Handler {
+    r := mux.NewRouter()
+    e := ep.MakeEndpoints(s)
+    options := []httptransport.ServerOption{}
+
+    // GET /books/:id                 Get a book by ID
+
+    r.Methods(http.MethodGet).Path("/books/{id}").Handler(httptransport.NewServer(
+        e.GetBook,
+        decodeGetBookRequest,
+        encodeResponse,
+        options...,
+    ))
+
+    return r
+}
+
+// decodeGetBookRequest decodes a request with the following format: GET /books/{id}
+func decodeGetBookRequest(_ context.Context, r *http.Request) (request interface{}, err error) {
+    vars := mux.Vars(r)
+    id, ok := vars["id"]
+    if !ok {
+        return nil, errors.New("Unable to get parameter, bad route")
+    }
+    return ep.GetBookRequest{ID: id}, nil
+}
+
+// errorer is implemented by all concrete response types that may contain
+// errors. It allows us to change the HTTP response code without needing to
+// trigger an endpoint (transport-level) error. For more information, read the
+// big comment in endpoints.go.
+type errorer interface {
```

Comment:
I think the big comment in `endpoints.go` is missing. But I guess how I read this is that business logic errors are encoded in the interface (so like a not found, etc), but a more technical/transport level exception, such as a resource not being available, bad argument, etc would be returned through the `error` of the `interface{}, error` pair?

Reply:
It's missing a general comment about how the content of `endpoints.go` work? 

I believe your read is correct. I'm new to this style of error handling. Normally, I'd have an encoded response for each endpoint. I borrowed this from go-kit examples, as it is a reusable method, similar to what we discussed the other day. The idea is to make sure to return an error string, not an interface.

Reply:
I mean the line `For more information, read the big comment in endpoints.go.` There isn't really a comment in `endpoints.go` that fits. Unless I'm missing something.

Reply:
Ahh, I see. I can add that comment to `endpoints.go`.